src/http_module.cpp

```diff
@@ -702,6 +718,40 @@ char* addResourceAttr(ngx_conf_t* cf, ngx_command_t* cmd, void* conf)
     return NGX_CONF_OK;
 }
 
+char *pemRootCertsRead(ngx_conf_t *cf, void *post, void *data) {
+    ngx_str_t *path = (ngx_str_t *)data;
+    auto mcf = getMainConf(cf);
+
+    if (path->len > 0) {
+        ngx_str_t full_path;
+        if (ngx_get_full_name(cf->pool, &full_path, &mcf->pemRootCerts) != NGX_OK) {
+            ngx_conf_log_error(NGX_LOG_EMERG, cf, 0, "Failed to get full path for %V", mcf->pemRootCerts);
+            return (char*)NGX_CONF_ERROR;
+        }
+
+        try {
+            std::ifstream file;
+            file.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+            file.open({(const char *)mcf->pemRootCerts.data, mcf->pemRootCerts.len}, std::ios::ate);
+            auto total_size = file.tellg();
+            file.seekg(0, std::ios::beg);
+
+            u_char *buf = (u_char *)ngx_palloc(cf->pool, total_size);
+            file.read((char *)buf, total_size);
+            mcf->pemRootCertsContent.data = buf;
+            mcf->pemRootCertsContent.len = total_size;
+
+            ngx_log_error(NGX_LOG_NOTICE, cf->log, 0, "Successfully read TLS certificate: %s", mcf->pemRootCerts.data);
+            return NGX_CONF_OK;
+        } catch (std::system_error& e) {
```

Comment:
It's usually good idea to catch the most basic exception to not miss anything. Also, `const`.